### PR TITLE
adding task 1 and 2

### PR DESCRIPTION
**Pull request by yasarosmankhan**
previous branch had too many issues https://github.com/yasarosmankhan/nc_news
created new one and copied code over with changes as instructed in 2 reviews

### CODE REVIEW

File: __tests__/index.test.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	});
+});
+
+describe('GET /api/invalid-path', () => {
+	test('GET:404 returns no content message', () => {
+		return request(app)
+			.get('/api/invalid-path')
+			.expect(404)
+			.then(({ body }) => {
+				expect(body.message).toBe('Not Found!');
```
Comment (B7Bprog) [inline __tests__/index.test.js]: This message could be something like 'wrong path' as this error is not related to missing specific data.

File: __tests__/index.test.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+			.then(({ body }) => {
+				const endpointsObj = body.endpoints;
+				for (const [endpoint, key] of Object.entries(endpointsObj)) {
+					expect(key.description).toEqual(expect.any(String));
+					if (key.queries !== undefined) {
+						expect(key.queries).toEqual(expect.any(Array));
+					}
+					if (key.exampleResponse !== undefined) {
+						expect(key.exampleResponse).toEqual(expect.any(Object));
+					}
```
Comment (B7Bprog) [inline __tests__/index.test.js:52]: A suggestion: Here you could also compare the whole object from the response to the one from the json file which could be required here in the test.

File: app.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const express = require('express');
+const app = express();
+const { getTopics, getEndPoints } = require('./controllers/topic.controllers');
+
+app.get('/api/topics', getTopics);
+
+app.get('/api', getEndPoints);
+
+app.all('/api/*', (req, res, next) => {
```
Comment (B7Bprog) [inline app.js]: This could be /* as it's a wrong path as well.

File: controllers/topic.controllers.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+const fs = require('fs/promises');
+const { selectTopics } = require('../models/topic.models');
+
+exports.getTopics = (req, res, next) => {
+	selectTopics()
+		.then((topics) => {
+			res.status(200).send({ topic: topics });
```
Comment (B7Bprog) [inline controllers/topic.controllers.js]: The topics should be send back on a key of 'topics' but currently they are on a key of 'topic'.

File: __tests__/index.test.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+beforeEach(() => seed(data));
+afterAll(() => db.end());
+
+describe('GET /api/topics', () => {
+	test('GET:200 return an array of topics of the correct format', () => {
+		return request(app)
+			.get('/api/topics')
+			.expect(200)
+			.then(({ body }) => {
+				expect(body.topic.length).toBe(3);
```
Comment (B7Bprog) [inline __tests__/index.test.js]: The topics should be send back on a key of 'topics' but currently they are on a key of 'topic'.

File: __tests__/index.test.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+					expect(topic).toMatchObject({
+						slug: expect.any(String),
+						description: expect.any(String),
+					});
+				});
+			});
+	});
+});
+
+describe('GET /api/invalid-path', () => {
```
Comment (B7Bprog) [inline __tests__/index.test.js:27]: Great job writing this test in a separate describe block.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Yasar, only a few minor things to change (see comments), then you can merge it. 👍

**Comment by yasarosmankhan** [comment]
approved
